build.rs

```diff
@@ -365,7 +391,7 @@ fn pregenerate_asm_main() {
             let srcs = asm_srcs(perlasm_src_dsts);
             for src in srcs {
                 let obj_path = obj_path(&pregenerated, &src, MSVC_OBJ_EXT);
-                run_command(nasm(&src, asm_target.arch, &obj_path, &pregenerated));
+                run_command(win_asm(&src, asm_target.arch, &obj_path, &pregenerated));
```

Comment:
It seems like this should be made consistent with the code below that does this:

```
let cmd = if target.os != WINDOWS || (ext != "asm" && ext != "S") {
            cc(p, ext, target, warnings_are_errors, &out_path, out_dir)
        } else {
            win_asm(p, &target.arch, &out_path, out_dir)
        };
```

Probably factor out this conditonal into a function (named `asm` perhaps) and use it here and below.

In particular, it is confusing how `win_asm` is used for `.S` files here (IIUC) but not used for `.S` files below.

Reply:
Disregard this. This was leftover feedback that was building on some other feedback I deleted before posting.